src/Raven.Server/NotificationCenter/Revisions.cs

```diff
@@ -0,0 +1,238 @@
+using Raven.Client.Documents.Conventions;
+
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Raven.Server.NotificationCenter.Notifications;
+using Sparrow.Logging;
+using Sparrow.Json;
+using System.Collections;
+
+namespace Raven.Server.NotificationCenter;
+public class Revisions
+{
+    private static readonly string ConflictRevisionExceededMaxId = "ConflictRevisionExceededMax";
+
+    private readonly NotificationCenter _notificationCenter;
+    private readonly NotificationsStorage _notificationsStorage;
+    private readonly string _database;
+
+    private readonly object _locker = new object();
+    private readonly DetailsUniqueQueue _queue = new DetailsUniqueQueue(maxSize: 64);
+    public enum ExceedingReason
+    {
+        MinimumRevisionAgeToKeep,
+        MinimumRevisionsToKeep
+    }
+
+    private Timer _timer;
+    private readonly Logger _logger;
+
+    public Revisions(NotificationCenter notificationCenter, NotificationsStorage notificationsStorage, string database)
+    {
+        _notificationCenter = notificationCenter;
+        _notificationsStorage = notificationsStorage;
+        _database = database;
+        _logger = LoggingSource.Instance.GetLogger(database, GetType().FullName);
+    }
+
+    public void Add(ConflictInfo info)
+    {
+        _queue.Enqueue(info);
+
+        if (_timer != null)
+            return;
+
+        lock (_locker)
+        {
+            if (_timer != null)
+                return;
+
+            _timer = new Timer(Update, null, TimeSpan.FromMinutes(1), TimeSpan.FromMinutes(1));
+        }
+    }
+
+    private void Update(object state)
+    {
+        try
+        {
+            if (_queue.IsEmpty)
+                return;
+
+            PerformanceHint hint = GetPagingPerformanceHint();
+            var details = ((ConflictPerformanceDetails)hint.Details);
+            while (_queue.TryDequeue(out ConflictInfo pagingInfo))
+            {
+                details.Update(pagingInfo);
+            }
+
+            _notificationCenter.Add(hint);
+        }
+        catch (Exception e)
+        {
+            if (_logger.IsInfoEnabled)
+                _logger.Info("Error in a notification center revisions timer", e);
+        }
+    }
+
+
+    private PerformanceHint GetPagingPerformanceHint()
+    {
+        using (_notificationsStorage.Read(ConflictRevisionExceededMaxId, out NotificationTableValue ntv))
+        {
+            ConflictPerformanceDetails details;
+            if (ntv == null || ntv.Json.TryGet(nameof(PerformanceHint.Details), out BlittableJsonReaderObject detailsJson) == false || detailsJson == null)
+                details = new ConflictPerformanceDetails();
+            else
+                details = DocumentConventions.DefaultForServer.Serialization.DefaultConverter.FromBlittable<ConflictPerformanceDetails>(detailsJson, ConflictRevisionExceededMaxId);
+
+            return PerformanceHint.Create(_database, "Excess number of Conflict Revisions",
```

Comment:
it is not a performance hint, it is an alert

Reply:
Done